docs/source/design/user_messaging.rst

```diff
@@ -0,0 +1,110 @@
+.. -*- mode: rst -*-
+.. vi: set ft=rst sts=4 ts=4 sw=4 et tw=79:
+
+.. _chap_design_user_messaging:
+
+*******************************************************
+User messaging: result records vs exceptions vs logging
+*******************************************************
+
+.. topic:: Specification scope and status
+
+   This specification provides a partial overview of the current implementation.
+
+Motivation
+==========
+
+This specification delineates the applicable contexts for using
+:ref:`result records <chap_design_result_records>`, :ref:`exceptions <chap_design_exception_handling>`,
+:ref:`progress reporting <chap_design_progress_reporting>`, specific :ref:`log levels <chap_design_log_levels>`,
+or other types of user messaging processes.
+
+
+Specification
+=============
+
+Result records
+--------------
+
+**Result records are the only return value format** for all DataLad interfaces.
```

Comment:
Please make it a little more specific (reference or description such as "subclasses of `Interface`") on what `DataLad interfaces` are